sdk/iOS/src/MSClient.h

```diff
@@ -140,22 +140,22 @@
 #if TARGET_OS_IPHONE
 /// Logs in the current end user with the given provider by presenting the
 /// MSLoginController with the given controller.
--(void)loginWithProvider:(NSString *)provider
-              controller:(UIViewController *)controller
+-(void)loginWithProvider:(nonnull NSString *)provider
+              controller:(nonnull UIViewController *)controller
```

Comment:
did you check this one?  I'm not sure if this is required or not off the top of my head